Migrate taleo to greenhouse

pmahnke opened the pull request:
## Done

* HR didn't want to use the greenhouse.io job listing page as the departments were incorrect, so using the api, we have replaced the older taleo page that will link to the hosted greenhouse job descriptions
* Robin added a new get_raw_json_feed template tag to read this format of json

## QA

- download this branch and `./run` the server
- look at http://0.0.0.0:8002/careers/all-vacancies and see that the page lists all open jobs, grouped by an external department field from greenhouse.io

## Issue / Card

Fixes #220 

## Screenshots

![image](https://user-images.githubusercontent.com/441217/33211815-2972c27e-d118-11e7-88bb-3da2b2eb9d60.png)

webteam-app commented:
Starting demo at: http://www.canonical.com-pr-221.run.demo.haus/

pmahnke commented:
this will go live on 11 December 2017

kwm14 commented:
## Visual review

Few visual tweaks :slightly_smiling_face:

### Summary page
- Update hero section to use `p-strip--image is-dark` class and inherit aubergine background as documented [here](https://docs.vanillaframework.io/en/patterns/strip#strip-image)
- Update department titles to use `p-heading--four`
- Update margin-top on Sales department to `16px` so it's consistent with other department styling
- Visual reference below with my quick hack

![screen shot 2017-11-24 at 14 34 29](https://user-images.githubusercontent.com/17748020/33214868-82cba07c-d125-11e7-9504-4f24c6a26f84.png)

### Job description page
- The styling me and Caleb implemented has changed? Looks as if it's reverted back to original styling before we made design tweaks to it...

anthonydillon commented on templates/careers/all-vacancies.html:
```diff
@@ -1,20 +1,56 @@
-      <p>If you are interested in a role in a specific department or location, you can <a href="https://ldd.tbe.taleo.net/ldd03/ats/careers/jobSearch.jsp?org=CANONICAL&amp;cws=1&amp;rid=86">search jobs</a>.</p>
-      {% get_rss_feed "http://ldd.tbe.taleo.net/ldd03/ats/servlet/Rss?org=CANONICAL&_rss_version=2&cws=1" as vacancies %} {% include "_includes/vacancies.html" with vacancies=vacancies %}
+      {% if jobs is False %}
+        <p class="u-no-margin--bottom">Feed failed to load. If you have time, please <a href="https://github.com/canonical-websites/www.canonical.com/issues/new">file an issue</a>.</p>
```
Maybe use an error notification?

anthonydillon commented on templates/careers/job.html:
```diff
@@ -0,0 +1,27 @@
+      {% get_raw_json_feed "https://api.greenhouse.io/v1/boards/Canonical/jobs/"|add:job_id  as listing %}
+
+{% if listing is False %}
+<p>got nothing</p>
```
Descriptive notification?

anthonydillon commented on templates/careers/job.html:
```diff
@@ -0,0 +1,27 @@
+<p>got nothing</p>
+{% else %}
+{% autoescape off %}
+{{ listing.content |decode }}
```
Remove the space before the pipe

anthonydillon commented on templates/careers/job.html:
```diff
@@ -0,0 +1,27 @@
+      <h1>All current vacancies</h1>
+      {% get_raw_json_feed "https://api.greenhouse.io/v1/boards/Canonical/jobs/"|add:job_id  as listing %}
+
+{% if listing is False %}
```
Can you indent please

pmahnke commented:
@kwm14 updated, however used p-strip--accent as --image had no background?

@anthonydillon removed job.html page as it is not needed ... added the error notification

![image](https://user-images.githubusercontent.com/441217/33228683-c156f29a-d1b8-11e7-9edb-9c97f227412b.png)

anthonydillon commented:
Cheers @pmahnke, I think the notification should be on the strip below the hero strip. Where the feed is outputted.

pmahnke commented:
@anthonydillon I knew you were going to say that... more logic!

pmahnke commented:
@anthonydillon 

* moved the notification of no results into the white content area
* added a check if the feed failed to return results

![image](https://user-images.githubusercontent.com/441217/33238123-542e47ce-d27e-11e7-948a-151be4acb3ff.png)

kwm14 commented:
@pmahnke looking good just one small update you missed:

- [ ] Update margin-top on Sales department to `16px` so it's consistent with other department styling

kwm14 commented:
Thanks @pmahnke 👍

anthonydillon commented:
This branch could to merged into a feature branch and create a go live card for next iteration.
